src/main.rs

```diff
@@ -481,6 +493,7 @@ mod tests {
             scan_order: ScanOrder::Serial,
             no_config: false,
             no_nmap: false,
+            top: false,
```

Comment:
Can you add another one of these tests and make `top: true`? I'm not sure if it'll work with our CI :shrug: but if it does that's great and we can maintain high test coverage :smile:

Reply:
on 2nd thoughts this would not work right now

Reply:
I have a plan to refactor the tests a little bit, but that will be better in another PR, because most of them have to change a little. Then we can make a new test for this functionality as well.